src/Scene/StageScene.cpp

```diff
@@ -0,0 +1,178 @@
+        NERVE_MAKE(StageScene, Shop);
+        NERVE_MAKE(StageScene, Closet);
+        NERVE_MAKE(StageScene, DemoShineGet);
+        NERVE_MAKE(StageScene, DemoTalk);
+        NERVE_MAKE(StageScene, Radicon);
+        NERVE_MAKE(StageScene, Warp);
+        NERVE_MAKE(StageScene, DemoGetLifeMaxUpItem);
+        NERVE_MAKE(StageScene, DemoWorldIntroCamera);
+        NERVE_MAKE(StageScene, DemoWorldIntroCameraBeforeAppearElectricDemo);
+        NERVE_MAKE(StageScene, SnapShot);
+        NERVE_MAKE(StageScene, DemoShineMainGet);
+        NERVE_MAKE(StageScene, DemoShineGrandGet);
+        NERVE_MAKE(StageScene, SkipDemo);
+        NERVE_MAKE(StageScene, Miss);
+        NERVE_MAKE(StageScene, DemoCarryMeat);
+        NERVE_MAKE(StageScene, RaceYukimaru);
+        NERVE_MAKE(StageScene, RaceYukimaruTutorial);
+        NERVE_MAKE(StageScene, RaceManRace);
+        NERVE_MAKE(StageScene, DemoTitleLogo);
+        NERVE_MAKE(StageScene, Pause);
+        NERVE_MAKE(StageScene, ArriveAtCheckpoint);
+        NERVE_MAKE(StageScene, TimeBalloon);
+        NERVE_MAKE(StageScene, DemoScenarioCamera);
+        NERVE_MAKE(StageScene, AppearFromWorldWarpHole);
+        NERVE_MAKE(StageScene, DemoStageStartOpening);
+        NERVE_MAKE(StageScene, DemoStageStart);
+        NERVE_MAKE(StageScene, DemoPlayerDown);
+        NERVE_MAKE(StageScene, DemoStageStartCapManHeroTalk);
+        NERVE_MAKE(StageScene, DemoCapManHeroTalkMoonRock);
+        NERVE_MAKE(StageScene, DemoReturnToHome);
+        NERVE_MAKE(StageScene, DemoHackStart);
+        NERVE_MAKE(StageScene, Play);
+        NERVE_MAKE(StageScene, DemoWithPlayerUseCoinNoFirst);
+        NERVE_MAKE(StageScene, DemoDamageStopPlayer);
+        NERVE_MAKE(StageScene, DemoSceneStartPlayerWalk);
+        NERVE_MAKE(StageScene, DemoRiseMapParts);
+        NERVE_MAKE(StageScene, DemoShineChipComplete);
+        NERVE_MAKE(StageScene, DemoHackStartFirst);
+        NERVE_MAKE(StageScene, DemoNormalBazookaElectric);
+        NERVE_MAKE(StageScene, DemoWithPlayerUseCoin);
+        NERVE_MAKE(StageScene, DemoWithPlayer);
+        NERVE_MAKE(StageScene, DemoNormal);
+        NERVE_MAKE(StageScene, Amiibo);
+        NERVE_MAKE(StageScene, DemoStageStartElectricDemo);
+        NERVE_MAKE(StageScene, WaitStartWarpForSession);
+        NERVE_MAKE(StageScene, DemoCountCoin);
+        NERVE_MAKE(StageScene, WarpToCheckpoint);
+        NERVE_MAKE(StageScene, DemoAppearFromHome);
+        NERVE_MAKE(StageScene, DemoRisePyramid);
+        NERVE_MAKE(StageScene, WaitWarpToCheckpoint);
+    } nrvStageScene;
+}
+
+namespace rs {
+    void endDemoCommonProc(al::Scene* scene, ProjectItemDirector* projItemDir);
+    bool isPlayerHackRadiconNpc(const al::LiveActor* actor);
+    void setKidsModeLayoutDisable(const al::IUseSceneObjHolder*);
+    void killCoinColletHintEffect(const al::IUseSceneObjHolder*);
+    void updateKitListDemoPlayerWithPauseEffect(al::Scene*);
+    bool isActiveDemo(al::Scene*);
+} // namespace rs
```

Comment:
no forward-declaring of functions

Reply:
i have no idea where to put these functions, so if you could suggest a spot, that would be great. lots of rs functions don't show up in the map properly with a place to put them, or sometimes they don't show up at all. `endDemoCommonProc` has no folder associated with it, and `isPlayerHackRadiconNpc` does not show up in the function map.

Reply:
That's a thing for @GRAnimated to fix the function map first.

Reply:
... so where do i put them? i have no reference of where rs functions are supposed to go in general.